conda/core/index.py

```diff
@@ -331,7 +333,62 @@ def maybe_decompress(filename, resp_content):
                              getattr(e.response, 'elapsed', None))
 
 
-def read_local_repodata(cache_path):
+def write_pickled_repodata(cache_path, repodata):
+    # Don't bother to pickle empty channels
+    if not repodata.get('packages'):
+        return
+    try:
+        with open(cache_path + '.q', 'wb') as f:
+            pickle.dump(repodata, f)
+    except:
+        pass
+
+
+def read_picked_repodata(cache_path, channel_url, schannel, priority):
+    pickle_path = cache_path + '.q'
+    # Don't trust pickled data if there is no accompanying json data
+    if not isfile(pickle_path) or not isfile(cache_path):
+        return None
+    try:
+        with open(cache_path + '.q', 'rb') as f:
+            repodata = pickle.load(f)
+    except:
+        return None
+    if repodata.get('_url') != channel_url:
+        return None
+    add_pip = context.add_pip_as_python_dependency
+    changed = False
+    # Safe changes to IndexRecord
+    if ((repodata['_add_pip'] != add_pip or
```

Comment:
Ugg the pip switching thing is just extra crud.  And in the larger context, it's unimportant.  The single most important part of this strategy is the ability to switch priority.  That's the one we need to keep.